### PR TITLE
WebUSB: Switch terminal for Xterm.js

### PR DESCRIPTION
**Pull request by xesxen**

### CODE REVIEW

**Comment by jorisplusplus** [comment]
Pasting doesn't seems to work with this new terminal
Almost have a change ready that fixes this issue

File: webusb_site/src/components/Programming.vue [hunk 1]
```diff
@@ -35,8 +35,7 @@
         <section>
               Python terminal
               <div class="md-form">
-                  <textarea placeholder=">>>" readonly wrap="hard" id="commandlog" style="resize: none; overflow:auto" v-model="commandlog"></textarea>
-                  <input autocomplete="off" type="text" id="example1" class="form-control" v-on:keyup="commandpython" v-model="command">
+                  <Terminal ref="terminal" v-on:key="commandpython"></Terminal>
```
Comment (jorisplusplus) [inline webusb_site/src/components/Programming.vue]:
```suggestion
                  <Terminal ref="terminal" v-on:data="commandpython"></Terminal>
```

File: webusb_site/src/components/Terminal.vue [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+                convertEol: true,
+            });
+            this.fitAddon = new FitAddon();
+            this.terminal.loadAddon(this.fitAddon);
+            this.terminal.open(this.$el);
+            this.terminal.write(">>> ");
+            this.fitAddon.fit();
+            window.addEventListener('resize', this.handleResize);
+
+            this.terminal.onKey((e) => {
```
Comment (jorisplusplus) [inline webusb_site/src/components/Terminal.vue]:
```suggestion
            this.terminal.onData((e) => {
```

File: webusb_site/src/components/Terminal.vue [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+                this.$emit('key', e.key);
```
Comment (jorisplusplus) [inline webusb_site/src/components/Terminal.vue]:
```suggestion
                this.$emit('data', e);
```

**Review by jorisplusplus — CHANGES_REQUESTED** [review]
Changing the onKey even listenener to onData will handle both typing and pasting.

**Review by jorisplusplus — APPROVED** [review]
Looks good to me.  Tom will probably check it later this day.

**Comment by tjclement** [comment]
Super cool addition! I will merge the change and make two new issues for this feature:

1. Backspace doesn't work in paste mode
2. Typing feels a bit sluggish because of the low frequency with which we poll the badge for new output.

**Pull request merged**